core/shared/src/main/scala/scodec/bits/ByteVector.scala

```diff
@@ -777,6 +778,37 @@ sealed abstract class ByteVector extends BitwiseOperations[ByteVector, Long] wit
     bldr.toString
   }
 
+  /**
+    * Converts the contents of this vector to a base 58 string.
+    *
+    * @group conversions
+    */
+  final def toBase58: String = toBase58(Bases.Alphabets.Base58)
+
+  /**
+    * Converts the contents of this vector to a base 64 string using the specified alphabet.
+    *
+    * @group conversions
+    */
+  final def toBase58(alphabet: Bases.Alphabet): String = {
+    val bytes = toArray
+    val ZERO = BigInt(0)
+    val RADIX = BigInt(58)
+    val ones = List.fill(toBase64.takeWhile(_ == 0).length)(1)
```

Comment:
This seems pretty expensive -- converting to base64 to determine the num er of leading 0s.

Reply:
Not sure why this is looking at `toBase64` at all -- I think this should just be `this.takeWhile(_ == 0).size`.